AMP-114937 update taxonomy api docs to support new delete/restore functionality (#374)
amplitude/amplitude-docs: content/collections/api/en/taxonomy.md

postman_link: 'https://www.postman.com/amplitude-dev-docs/workspace/amplitude-developers/folder/20044411-4b8180c4-e06e-478a-9c96-0ccf6cf652e2?action=share&source=copy-link&creator=29131806&ctx=documentation'
api_status: ga
lede: |-

>   The Taxonomy API grants `Enterprise` users the ability to programmatically plan their event schema in the Taxonomy tab.

Reviewer: ```suggestion
  The Taxonomy API grants Enterprise users the ability to programmatically plan their event schema in the Taxonomy tab.
```

No need to add code formatting to the word Enterprise

Author: Some of our customers (who are not on an enterprise plan) kinda missed this part and raised some tickets related to this and I thought highlighting the plan name will at least reduce such instances.

{{/partial:tabs}}

{{partial:collapse name="Example: Get an event type by name"}}

> This example gets the "Event 2" event type.

Reviewer 2: ```suggestion
This example gets the 'Event 2' event type.
```

{{/partial:tabs}}

{{partial:collapse name="Example: Update an event type"}}

> This example updates the event type "OnboardingBegin" with the category "Onboarding", event type name "OnboardStart", the display name "Onboarding Start", and a description of "User signed in and completed an onboarding task from modal".

Reviewer 2: ```suggestion
This example updates the event type `OnboardingBegin` with the category `Onboarding`, event type name `OnboardStart`, the display name "Onboarding Start", and a description of "User signed in and completed an onboarding task from modal".
```

|<div class="big-column">Name</div>| Description|
|----|-----|
|`event_type`| <span class="required">Required</span>. The name of the event type.|

#### Behavior

> 

Reviewer: Let's make this list a two column table

}
```

#### 4XX responses

> If there is a problem with your request, the request returns a `4XX` status, and a JSON body with more information.

Reviewer: ```suggestion
If there's a problem with the request, Amplitude returns a `4XX` status, and a JSON body with more information.
```

Host: amplitude.com
Authorization: Basic {api-key}:{secret-key} #credentials must be base64 encoded
```
{{/partial:tab}}
{{/partial:tabs}}

{{partial:collapse name="Example: Restore an event type"}}
This example restores the event type "Event1".

{{partial:tabs tabs="cURL, HTTP"}}
{{partial:tab name="cURL"}}
```bash
curl --location --request POST 'https://amplitude.com/api/2/taxonomy/event/Event1/restore' \
--header 'Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA='

```
{{/partial:tab}}
{{partial:tab name="HTTP"}}
```bash
POST /api/2/taxonomy/event/Event1/restore HTTP/1.1
Host: amplitude.com
Authorization: Basic MTIzNDU2NzgwMDoxMjM0NTY3MDA=

```
{{/partial:tab}}
{{/partial:tabs}}
{{/partial:collapse}}

#### Path parameters

|<div class="big-column">Name</div>| Description|
|----|-----|
|`event_type`| <span class="required">Required</span>. The name of the event type.|

#### Behavior

* If the event type is `deleted`, it is restored.
* If the event type is `not deleted` or `not found`, an error is returned.

#### 200 OK response

A successful request returns a `200 OK` status and a JSON body.

```json
{
    "success": true
}
```

#### 4XX responses

> If there is a problem with your request, the request returns a `4XX` status, and a JSON body with more information.

Reviewer: ```suggestion
If there's a problem with your request, Amplitude returns a `4XX` status, and a JSON body with more information.
```

|<div class="big-column">Name</div>|Description|
|-----|---------|
|`event_type`|<span class="optional">Optional</span>. Name of the event type to which the event properties belong to.|

#### Behavior

> 

Reviewer: Similar to above, this should be more readable as a table

|-----|---------|
|`user_property`|<span class="required">Required</span>. The user property name. Prefix custom user properties with `gp:`.|

#### Behavior

> 

Reviewer: Let's do a table here.
